internal/validators/namespace_test.go

```diff
@@ -335,3 +335,65 @@ func TestIllegalIncludedNamespaceNamespace(t *testing.T) {
 		})
 	}
 }
+
+func TestNsTreeLabelAddition(t *testing.T) {
+
+	f := foresttest.Create("-")
+	vns := &Namespace{Forest: f}
+
+	ns := &corev1.Namespace{}
+	ns.Name = "a"
+
+	ns.SetLabels(map[string]string{api.LabelTreeDepthSuffix: "1"})
+
+	tests := []struct {
+		name string
+		op   k8sadm.Operation
+		fail bool
+	}{
+		{name: "Creation of namespace with tree label in it", op: k8sadm.Create, fail: false},
+		{name: "Update namespace with tree label in it", op: k8sadm.Update, fail: false},
+	}
+	for _, tc := range tests {
+		g := NewWithT(t)
+		req := &nsRequest{
+			ns: ns,
+			op: tc.op,
+		}
+		got := vns.illegalTreeLabel(req)
+
+		logResult(t, got.AdmissionResponse.Result)
+		g.Expect(got.AdmissionResponse.Allowed).Should(Equal(tc.fail))
+	}
+
+}
+
+func TestNsUpdateValueTreeLabel(t *testing.T) {
```

Comment:
You should be able to combine this with the last test just by changing the op (and not passing in `oldNS` if the op is `Create`)